File: 백준/Silver/2606. 바이러스/바이러스.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const totalComputer = Number(input.shift());
const pairComputer = Number(input.shift());
const computers = input.map((v) => v.split(" ").map(Number));
let answer = 0;
let visited = Array(totalComputer + 1).fill(false);
let graph = [...Array(totalComputer + 1)].map(() => []);

computers.map(([from, to]) => {
  graph[from].push(to);
  graph[to].push(from);
});

const bfs = (start) => {
  let queue = [start];
  while (queue.length) {
    let node = queue.shift();
    if (visited[node]) continue;
    else {
      visited[node] = true;
      answer++;
      queue.push(...graph[node]);
    }
  }
};
bfs(1);
console.log(answer - 1);

File: 백준/Silver/2606. 바이러스/바이러스.py
n=int(input())
m=int(input())
graph=[[]*n for _ in range(n+1)]
visited=[0]*(n+1)

for _ in range(m):
    a,b=map(int,input().split())
    graph[a].append(b)
    graph[b].append(a)
    graph[a].sort()

def dfs(node):
    visited[node]=1
    for i in graph[node]:
        if visited[i]==0:
            dfs(i)
    return
dfs(1)
print(sum(visited)-1)